src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -209,6 +209,7 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     //
                     // Avoid instance state modification
                     Dictionary<string, ConfigurationSetting> applicationSettings = null;
+                    Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings = null;

```

Comment:
Should be cleared at the beginning of ExecuteWithFailOverPolicyAsync like `applicationSettings` is. The reason is that if it's not it's possible a failover could see half-modified state of previous attempt.